src/doc/contrib/src/architecture/index.md

```diff
@@ -0,0 +1 @@
+# Architecture Overview
```

Comment:
Stray page?

Reply:
Not really stray, mdbook needs a page for nested chapters.  I considered adding some high-level overview, but couldn't really think of what to add. I could also just add like a table-of-contents for the sub-chapter?

Reply:
Nah that's fine, I think mdbook will render this though and you can click on it on the left-hand-side navigation to see an empty page, so perhaps just a small blurb here to lay out what's contained within?

Reply:
Yea, sounds good!  I added a short intro.